src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -281,18 +281,13 @@ private void AddSemanticRange(SyntaxNode node, int semanticKind)
             }
         }
 
-        private void AddSemanticRange(SyntaxNode node, SyntaxKind? kind = null)
+        private void AddSemanticRange(SyntaxNode node, SyntaxKind kind)
         {
             if (node is null)
             {
                 throw new ArgumentNullException(nameof(node));
             }
 
-            if (kind is null)
-            {
-                kind = node.Kind;
```

Comment:
Why not always use `node.Kind`? Not sure I understand why we need to pass in `kind` separately.

Reply:
So, in some cases we're passing a different SyntaxKind than the node reports (I believe RazorDirectiveBody.Keyword is one instance and VisitMarkupTagHelperDirectiveAttribute.Name is another) because we want to differentiate based on the position. I therefore figured that it was better to be explicit about how we wanted to treat every node in case there were more edge-cases in the syntax-tree which would give us different SyntaxKinds than we handle.

Reply:
After talking with the person being affected by this I also suspect that the problem they're having here is that their SyntaxTree is "off the rails" for a few seconds while they're editing, by asserting that the node in a given slot is of the type that  we  expect we can hopefully avoid this stuff.

Reply:
My concern is that we're not really asserting that the node has the kind it says it has but rather ignoring what the node says and applying the kind we expect it to. I understand that this could fix any issues due to the syntax tree being "off the rails" but if that is indeed the case, we need to fix the syntax tree itself. Is there a repro for when the syntax tree isn't what we expect?

Reply:
You're correct that we're not "asserting" so much as "overriding", that was poor wording on my part. I don't have a repro for the "off the rails" scenario, it's only speculation based on what I heard from our unfortunate colleague.

Unfortunately I've now been pulled into working on DDRIT reliability ASAP. Would you be OK if, should we do another insertion, I merged this PR as-is so we can stop yellow-baring people then file a follow-up issue to figure out what's up with their scenario?

Reply:
Yup. Totally fine merging it as is to get that unblocked 👍 . I just didn't want this hiding underlying bugs

Reply:
So the odd part about this is that we later translate SyntaxKind's into semantic kinds. Wouldn't that effectively be the same?